connection-limit/connection_limit_kern.c

```diff
@@ -273,17 +273,20 @@ int _xdp_limit_conn(struct xdp_md *ctx)
     if (data + sizeof(struct ethhdr)+ 1 > data_end)
         return XDP_PASS;
 
-    /* Check if its valid ip packet */
+    
     struct iphdr *iph = (struct iphdr *)(data + sizeof(struct ethhdr));
-    if (iph + 1 > data_end)
+    __u8 l4_offset = iph->ihl * 4; // ip header length
```

Comment:
Minimum value of IHL is 5. Add a check to discard invalid IP packets with IHL less than 5.